client/client_test.go

```diff
@@ -145,6 +146,63 @@ func (cs *clientSuite) TestClientWorks(c *C) {
 	c.Check(cs.req.URL.Path, Equals, "/this")
 }
 
+func (cs *clientSuite) TestClientSetMaintenanceForMaintenanceJSON(c *C) {
+	// write a maintenance.json that says snapd is down for a restart
+	maintErr := &client.Error{
+		Kind:    client.ErrorKindSystemRestart,
+		Message: "system is restarting",
+	}
+
+	b, err := json.Marshal(maintErr)
+	c.Assert(err, IsNil)
+
+	c.Assert(os.MkdirAll(filepath.Dir(dirs.SnapdMaintenanceFile), 0755), IsNil)
+	c.Assert(ioutil.WriteFile(dirs.SnapdMaintenanceFile, b, 0644), IsNil)
+	var v []int
+	cs.rsp = `[1,2]`
+	reqBody := ioutil.NopCloser(strings.NewReader(""))
+	statusCode, err := cs.cli.Do("GET", "/this", nil, reqBody, &v, nil)
+	c.Check(err, IsNil)
+	c.Check(statusCode, Equals, 200)
+	c.Check(v, DeepEquals, []int{1, 2})
```

Comment:
Do we test the deails about "Do()" here to ensure that even with the maintenance file in place we do the normal "do()" things?

Reply:
I will add this test

Reply:
I was more wondering about the reverse, we test a lot of details about "Do()" here when we are (it seems to me) mostly interested in the part of the test that details with the Maintenance() reply. Was mostly wondering if we should test less about Do() here :)

Reply:
ah sure fair enough we can drop those bits too